htdocs/product/class/api_products.class.php

```diff
@@ -173,9 +173,10 @@ public function getByBarcode($barcode, $includestockdata = 0, $includesubproduct
 	 * @param  int    $variant_filter   	Use this param to filter list (0 = all, 1=products without variants, 2=parent of variants, 3=variants only)
 	 * @param  bool   $pagination_data   	If this parameter is set to true the response will include pagination data. Default value is false. Page starts from 0
 	 * @param  int    $includestockdata		Load also information about stock (slower)
+	 * @param  bool   $allow_extrafields_sqlfilters Allow sqlfilters to contain extrafields filter (slower).
 	 * @return array                		Array of product objects
 	 */
-	public function index($sortfield = "t.ref", $sortorder = 'ASC', $limit = 100, $page = 0, $mode = 0, $category = 0, $sqlfilters = '', $ids_only = false, $variant_filter = 0, $pagination_data = false, $includestockdata = 0)
+	public function index($sortfield = "t.ref", $sortorder = 'ASC', $limit = 100, $page = 0, $mode = 0, $category = 0, $sqlfilters = '', $ids_only = false, $variant_filter = 0, $pagination_data = false, $includestockdata = 0, $allow_extrafields_sqlfilters = false)
```

Comment:
Because extrafields are now considered as "columns" of the same object, i think we don't need to add an option for this:
Just do the join always (cost is negligeable since link is done using a primary key).

Reply:
Done.